Sources/Player/PlaybackEngine/Internal/AVQueuePlayer/AVQueuePlayerWrapper.swift

```diff
@@ -184,13 +189,30 @@ final class AVQueuePlayerWrapper: GenericMediaPlayer {
 
 			self.delegates.seeking(in: asset)
 
-			let completed = await self.player.seek(to: time)
-			guard completed, currentItem == self.player.currentItem, self.player.timeControlStatus == .playing else {
-				return
-			}
+			if self.shouldPauseAndPlayAroundSeek {
+				if self.player.timeControlStatus == .playing {
+					self.isSeeking = true
+					await self.player.pause()
+				}
 
-			asset.setAssetPosition(currentItem)
-			self.delegates.playing(asset: asset)
+				let completed = await self.player.seek(to: time)
+				self.isSeeking = false
+				await self.player.play()
```

Comment:
From reading the documentation, seek should only fail if it is interrupted by other seek or similar operation.

When interrupted, the first seek finishes with completed = false and the next seek will finish with true.

In our case, because we have all the AVPlayer operations synchronous with the queue, I don't think it should return false.. so I think handling it like this is fine.

Reply:
yes, for testing I did the two seeks, but this scenario shouldn't ever happen so it was not "real", so I don't know how a real scenario would actually change the internal state which could - I don't know - possibly affect our current conditions.